src/components/Semester/Semester.js

```diff
@@ -1,16 +1,26 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
```

Comment:
useEffect hook is redundant in this file.
```suggestion
import React, { useState } from 'react';
```